src/main/java/io/github/externschool/planner/config/SpringSecurityConfig.java

```diff
@@ -7,28 +7,41 @@
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
 import org.springframework.security.crypto.password.NoOpPasswordEncoder;
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
 
 @Configuration
 @EnableWebSecurity
 public class SpringSecurityConfig extends WebSecurityConfigurerAdapter {
 
     @Override
     protected void configure(HttpSecurity http) throws Exception {
-        http.csrf().disable().authorizeRequests().anyRequest().fullyAuthenticated()
-                .antMatchers("/").permitAll()
+        http.csrf().disable()
+                .authorizeRequests()
+                .antMatchers("/", "/index", "/home", "/greeting",
+                        "/signup",
+                        "/js/**",
+                        "/css/**",
+                        "/img/**",
+                        "/webjars/**").permitAll()
+                .anyRequest().authenticated()
                 .and()
-                .formLogin().defaultSuccessUrl("/", true)
-                .loginPage("/login").permitAll().and().logout().permitAll()
-                .and().httpBasic();
-
+                .formLogin()
+                .loginPage("/login")
+                .permitAll()
+                .and()
+                .logout()
+                .invalidateHttpSession(true)
+                .clearAuthentication(true)
+                .logoutRequestMatcher(new AntPathRequestMatcher("/logout"))
+                .logoutSuccessUrl("/login?logout")
```

Comment:
We also don't have handler for urls `login` and `logout`
This handler is part of another issue.

For this case it's can be, but for the future please don't do that. Don't add code for the future